### ISSUE TITLE
Flaky test: TestServicePlugin

### ISSUE THREAD

**Issue by thaJeztah** [moby/moby#37132]
This has been failing on several PR's

- https://github.com/moby/moby/pull/36906 (https://github.com/moby/moby/pull/36906#issuecomment-385516999)
- https://github.com/moby/moby/pull/36944 (https://github.com/moby/moby/pull/36944#issuecomment-384735408)
- others that I didn't register 😅 

https://github.com/moby/moby/pull/37086 was increasing the timeout for swarm tests (but that was removed in favor of changing the timeout on a test-by-test base (https://github.com/moby/moby/pull/37086#discussion_r189062007))

```
00:49:31 === RUN   TestServicePlugin
00:50:30 --- FAIL: TestServicePlugin (59.53s)
00:50:30 	daemon.go:285: [d1e7868f8b0ee] waiting for daemon to start
00:50:30 	daemon.go:317: [d1e7868f8b0ee] daemon started
00:50:30 	daemon.go:275: [d1e7868f8b0ee] exiting daemon
00:50:30 	daemon.go:285: [d828dcf4073c4] waiting for daemon to start
00:50:30 	daemon.go:317: [d828dcf4073c4] daemon started
00:50:30 	daemon.go:285: [d409f87ad25cc] waiting for daemon to start
00:50:30 	daemon.go:317: [d409f87ad25cc] daemon started
00:50:30 	daemon.go:285: [d3b90e73c9588] waiting for daemon to start
00:50:30 	daemon.go:317: [d3b90e73c9588] daemon started
00:50:30 	plugin_test.go:92: timeout hit after 30s: plugin "test" exists
00:50:30 	daemon.go:275: [d3b90e73c9588] exiting daemon
00:50:30 	daemon.go:275: [d409f87ad25cc] exiting daemon
00:50:30 	daemon.go:275: [d828dcf4073c4] exiting daemon
00:50:30 FAIL
```

**Comment by arm64b** [comment]
Adjust the timeout value for this specific case or bump all the service timeout value? I don't think we leave this test case failure with timeout reason does make sense to help us to figure out the performance issue, as I said, couple of factors including the logic complexity of the test case itself, system performance, density of workload when the test begin to run and the scheduling...
Do we have some track system to record those slowdown test case and have plan to investigate it?

**Comment by arm64b** [comment]
https://jenkins.dockerproject.org/job/Docker-PRs/49551/console

> 01:40:58 === RUN   TestCreateServiceSecretFileMode
01:41:31 --- FAIL: TestCreateServiceSecretFileMode (32.33s)
01:41:31 	daemon.go:285: [d43073d4909b7] waiting for daemon to start
01:41:31 	daemon.go:317: [d43073d4909b7] daemon started
01:41:31 	create_test.go:190: timeout hit after 30s: task count at 1 waiting for 0
01:41:31 	daemon.go:275: [d43073d4909b7] exiting daemon

**Comment by arm64b** [comment]
Understood. Now we have captured 2 service related test cases timeout: `TestServicePlugin` and `TestCreateServiceSecretFileMode`.

**Comment by vdemeester** [comment]
> I don't think we have, currently, but having detailed metrics about tests (pass/failure to identify flaky ones, ones that fail on specific configurations, and time to complete) has been discussed frequently; don't think anyone had time yet to dive into that

Yep, it's exactly that : we discussed (for a while now :stuck_out_tongue_closed_eyes:), but so far didn't had time to dive into it :sob:

**Comment by olljanat** [comment]
PowerPC on #38050 failed to this:
```
18:02:16 === RUN   TestCreateServiceSecretFileMode
18:02:48 --- FAIL: TestCreateServiceSecretFileMode (32.66s)
18:02:48     daemon.go:296: [d7533b76d05d6] waiting for daemon to start
18:02:48     daemon.go:328: [d7533b76d05d6] daemon started
18:02:48     create_test.go:296: timeout hit after 30s: task count at 1 waiting for 0
18:02:48     daemon.go:283: [d7533b76d05d6] exiting daemon
```
